src/hooks/useResources.ts

```diff
@@ -1,513 +1,223 @@
-            type: PayloadTypes.sortingMove,
-            payload: {
-              currentMoves: data?.mapped || [],
-              baseMoves: data?.base,
-              currentPage: 1,
-              sorting,
-            },
-          });
-        } catch (error) {
-          handleCatch(error);
-        }
-      } else {
-        const sorted = sort(sorting as MoveSorting) as PokemonMoveDetails[];
-        const currentMoves = sorted?.slice(0, resultsAmount.moves);
-        dispatch({
-          type: PayloadTypes.sortingMove,
-          payload: {
-            sorting: sorting as MoveSorting,
-            searchResults: sorted,
-            currentPage: 1,
-            currentMoves,
-          },
-        });
-      }
+  };
+
+  const handleNextMovePage = () => {
+    if (moveSorting === MoveSorting.none && baseMoves?.next && !searchMoveResults?.length) {
+      dispatch(nextPageAsync({ resourceType: 'moves', next: baseMoves.next }));
+    } else {
+      dispatch(nextPageSync({ resourceType: 'moves' }));
     }
-    if (isType) {
-      if (sorting === TypeSorting.none && !searchResults.types?.length) {
-        const data = await fetchAndMapTypes(`${API.TYPE}?limit=${resultsAmount.types}`);
-        dispatch({
-          type: PayloadTypes.sortingType,
-          payload: {
-            currentTypes: data?.mapped || [],
-            baseTypes: data?.base,
-            currentPage: 1,
-            sorting,
-          },
-        });
-      } else {
-        const sorted = sort(sorting as TypeSorting) as PokemonTypeDetails[];
-        const currentTypes = sorted?.slice(0, resultsAmount.types);
-        dispatch({
-          type: PayloadTypes.sortingType,
-          payload: {
-            sorting: sorting as TypeSorting,
-            searchResults: sorted,
-            currentPage: 1,
-            currentTypes,
-          },
-        });
-      }
+  };
+
+  const handlePreviousMovePage = () => {
+    if (moveSorting === MoveSorting.none && baseMoves?.previous && !searchMoveResults?.length) {
+      dispatch(previousPageAsync({ resourceType: 'moves', previous: baseMoves.previous }));
+    } else {
+      dispatch(previousPageSync({ resourceType: 'moves' }));
     }
-    setIsLoading(false);
   };
 
-  const handleResultsAmount = async (resultsAmount: number) => {
-    setIsLoading(true);
-    if (isPokemon) {
-      if (shouldFetchSearch.pokemons) {
-        try {
-          const data = await fetchAndMapPokemons(`${API.POKEMON}?limit=${resultsAmount}`);
-          dispatch({
-            type: PayloadTypes.pokemonResults,
-            payload: {
-              currentPokemons: data?.mapped || [],
-              basePokemons: data?.base,
-              currentPage: 1,
-              resultsAmount,
-            },
-          });
-        } catch (error) {
-          handleCatch(error);
-        }
+  const handleSpecificMovePage = (event: MouseEvent<HTMLButtonElement>) => {
+    const page = Number(event.currentTarget.textContent);
```

Comment:
Do you think it is safe to manage page this way?

Reply:
Yeah, I quess. At the moment I can't imagine error situation where page would NaN. But even if page would be a NaN then state does not change and nothing happens.